From b1304c04917420803016a0b51a23ba2e4c4725b7 Mon Sep 17 00:00:00 2001
From: Small Bricktory <studio@smallbricktory.com>
Date: Wed, 28 Oct 2026 12:53:51 -0700
Subject: [PATCH] MobileVLCKit 3.7.4-bop2: remove GPL-licensed code, iOS minimum 26.0

- contrib bootstrap --disable-zvbi and VLC configure --disable-zvbi
  (libzvbi has GPL-2-only files; teletext goes through telx)
- blacklist the GPL-2+ modules rotate, vod_rtsp, stream_out_rtp,
  syslog, mono, dolby_surround_decoder, dummy (control)
- VLC patch 0047: remove the GPL-2+ yadif deinterlace modes
- VLC patch 0048: FFmpeg contrib without the GPL-2+ aarch64 AC-3 DSP
- SDK_MIN 26.0
- ASMFLAGS exported so CMake-built assembly objects (mpg123) also target iOS 26.0
- iOS simulator slice arm64 only (no x86_64)
---
 buildMobileVLCKit.sh                          |  19 +-
 ...ve-the-GPL-2-yadif-and-yadif2x-modes.patch | 884 ++++++++++++++++++
 ...o-not-build-the-GPL-2-aarch64-AC-3-D.patch |  72 ++
 3 files changed, 968 insertions(+), 7 deletions(-)
 create mode 100644 libvlc/patches/0047-deinterlace-remove-the-GPL-2-yadif-and-yadif2x-modes.patch
 create mode 100644 libvlc/patches/0048-contrib-ffmpeg-do-not-build-the-GPL-2-aarch64-AC-3-D.patch

diff --git a/buildMobileVLCKit.sh b/buildMobileVLCKit.sh
index 3e274a0..1e0eb1b 100755
--- a/buildMobileVLCKit.sh
+++ b/buildMobileVLCKit.sh
@@ -9,7 +9,7 @@ BUILD_SIMULATOR=yes
 BUILD_DYNAMIC_FRAMEWORK=no
 BUILD_ARCH=`uname -m | cut -d. -f1`
 SDK_VERSION=`xcrun --sdk iphoneos --show-sdk-version`
-SDK_MIN=9.0
+SDK_MIN=26.0
 VERBOSE=no
 DEBUG=no
 CONFIGURATION="Release"
@@ -237,7 +237,7 @@ buildxcodeproj()
         fi
         if [ "$IOS" = "yes" ]; then
             if [ "$PLATFORM" = "iphonesimulator" ]; then
-                architectures="arm64 x86_64"
+                architectures="arm64"
             else
                 if [ "$INCLUDE_32BIT" = "yes" ]; then
                     architectures="armv7 armv7s arm64"
@@ -437,6 +437,7 @@ buildLibVLC() {
     export CPPFLAGS="${EXTRA_CFLAGS}"
     export CXXFLAGS="${EXTRA_CFLAGS}"
     export OBJCFLAGS="${EXTRA_CFLAGS}"
+    export ASMFLAGS="${EXTRA_CFLAGS}"
     export LDFLAGS="${EXTRA_LDFLAGS}"
 
     spushd ${VLCROOT}/contrib
@@ -554,7 +555,7 @@ buildLibVLC() {
         --disable-sdl \
         --disable-SDL_image \
         --disable-iconv \
-        --enable-zvbi \
+        --disable-zvbi \
         --disable-kate \
         --disable-caca \
         --disable-gettext \
@@ -642,7 +643,7 @@ buildLibVLC() {
     fi
 
     if [ "$SCARY" = "yes" ]; then
-        SCARYFLAG="--enable-dvbpsi --enable-avcodec"
+        SCARYFLAG="--enable-dvbpsi --enable-avcodec --disable-zvbi"
     else
         SCARYFLAG="--disable-dca --disable-dvbpsi --disable-avcodec --disable-avformat --disable-zvbi --enable-vpx"
     fi
@@ -793,6 +794,13 @@ buildLibVLC() {
     fingerprinter
     output_udp
     output_livehttp
+    librotate_plugin
+    libvod_rtsp_plugin
+    libstream_out_rtp_plugin
+    libsyslog_plugin
+    libmono_plugin
+    libdolby_surround_decoder_plugin
+    libdummy_plugin
     "
 
     if [ "$SCARY" = "no" ]; then
@@ -880,7 +888,6 @@ buildMobileKit() {
             fi
             if [ "$IOS" = "yes" ]; then
                 if [ "$PLATFORM" = "iphonesimulator" ]; then
-                    buildLibVLC "x86_64" "Simulator"
                     buildLibVLC "aarch64" "Simulator"
                 else
                     if [ "$INCLUDE_32BIT" = "yes" ]; then
@@ -951,8 +958,6 @@ collect_symbols_and_libraries() {
     arch="aarch64"
     if [ "$FARCH" != "all" ];then
         arch="$FARCH"
-    elif [ "$BUILD_SIMULATOR" = "yes" ]; then
-        arch="x86_64"
     fi
 
     actual_arch=`get_actual_arch $arch`
diff --git a/libvlc/patches/0047-deinterlace-remove-the-GPL-2-yadif-and-yadif2x-modes.patch b/libvlc/patches/0047-deinterlace-remove-the-GPL-2-yadif-and-yadif2x-modes.patch
new file mode 100644
index 0000000..e5b32d0
--- /dev/null
+++ b/libvlc/patches/0047-deinterlace-remove-the-GPL-2-yadif-and-yadif2x-modes.patch
@@ -0,0 +1,884 @@
+From 2faf1fc9ec975c564f4a967896c278922d8a252f Mon Sep 17 00:00:00 2001
+From: Small Bricktory <studio@smallbricktory.com>
+Date: Wed, 28 Oct 2026 11:10:02 -0700
+Subject: [PATCH] deinterlace: remove the GPL-2+ yadif and yadif2x modes
+
+yadif.h and yadif_template.h come from MPlayer/FFmpeg under GPL-2+.
+Drop them and their LGPL wrapper algo_yadif.c/.h from the deinterlace
+plugin, and remove yadif/yadif2x from every deinterlace mode list
+(plugin, core option, vout, libvlc_video_set_deinterlace). The default
+mode auto maps to x, which stays.
+---
+ lib/video.c                                   |   1 -
+ modules/video_filter/Makefile.am              |   2 -
+ modules/video_filter/deinterlace/algo_yadif.c | 202 -------------
+ modules/video_filter/deinterlace/algo_yadif.h |  94 ------
+ .../video_filter/deinterlace/deinterlace.c    |   4 -
+ .../video_filter/deinterlace/deinterlace.h    |   5 +-
+ modules/video_filter/deinterlace/yadif.h      | 153 ----------
+ .../video_filter/deinterlace/yadif_template.h | 282 ------------------
+ src/libvlc-module.c                           |   4 +-
+ src/video_output/interlacing.c                |   2 -
+ 10 files changed, 4 insertions(+), 745 deletions(-)
+ delete mode 100644 modules/video_filter/deinterlace/algo_yadif.c
+ delete mode 100644 modules/video_filter/deinterlace/algo_yadif.h
+ delete mode 100644 modules/video_filter/deinterlace/yadif.h
+ delete mode 100644 modules/video_filter/deinterlace/yadif_template.h
+
+diff --git a/lib/video.c b/lib/video.c
+index 5b8db36fa..1e38a5a14 100644
+--- a/lib/video.c
++++ b/lib/video.c
+@@ -675,7 +675,6 @@ void libvlc_video_set_deinterlace( libvlc_media_player_t *p_mi, int deinterlace,
+      && strcmp (psz_mode, "blend")    && strcmp (psz_mode, "bob")
+      && strcmp (psz_mode, "discard")  && strcmp (psz_mode, "linear")
+      && strcmp (psz_mode, "mean")     && strcmp (psz_mode, "x")
+-     && strcmp (psz_mode, "yadif")    && strcmp (psz_mode, "yadif2x")
+      && strcmp (psz_mode, "phosphor") && strcmp (psz_mode, "ivtc")
+      && strcmp (psz_mode, "auto"))
+         return;
+diff --git a/modules/video_filter/Makefile.am b/modules/video_filter/Makefile.am
+index f65344d48..c1d8d08e3 100644
+--- a/modules/video_filter/Makefile.am
++++ b/modules/video_filter/Makefile.am
+@@ -131,8 +131,6 @@ libdeinterlace_plugin_la_SOURCES = \
+ 	video_filter/deinterlace/helpers.c video_filter/deinterlace/helpers.h \
+ 	video_filter/deinterlace/algo_basic.c video_filter/deinterlace/algo_basic.h \
+ 	video_filter/deinterlace/algo_x.c video_filter/deinterlace/algo_x.h \
+-	video_filter/deinterlace/algo_yadif.c video_filter/deinterlace/algo_yadif.h \
+-	video_filter/deinterlace/yadif.h video_filter/deinterlace/yadif_template.h \
+ 	video_filter/deinterlace/algo_phosphor.c video_filter/deinterlace/algo_phosphor.h \
+ 	video_filter/deinterlace/algo_ivtc.c video_filter/deinterlace/algo_ivtc.h
+ # inline ASM doesn't build with -O0
+diff --git a/modules/video_filter/deinterlace/algo_yadif.c b/modules/video_filter/deinterlace/algo_yadif.c
+deleted file mode 100644
+index a89d240b8..000000000
+--- a/modules/video_filter/deinterlace/algo_yadif.c
++++ /dev/null
+@@ -1,202 +0,0 @@
+-/*****************************************************************************
+- * algo_yadif.c : Wrapper for FFmpeg's Yadif algorithm
+- *****************************************************************************
+- * Copyright (C) 2000-2011 VLC authors and VideoLAN
+- * $Id$
+- *
+- * Author: Laurent Aimar <fenrir@videolan.org>
+- *         Juha Jeronen  <juha.jeronen@jyu.fi> (soft field repeat hack)
+- *
+- * This program is free software; you can redistribute it and/or modify it
+- * under the terms of the GNU Lesser General Public License as published by
+- * the Free Software Foundation; either version 2.1 of the License, or
+- * (at your option) any later version.
+- *
+- * This program is distributed in the hope that it will be useful,
+- * but WITHOUT ANY WARRANTY; without even the implied warranty of
+- * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+- * GNU Lesser General Public License for more details.
+- *
+- * You should have received a copy of the GNU Lesser General Public License
+- * along with this program; if not, write to the Free Software Foundation,
+- * Inc., 51 Franklin Street, Fifth Floor, Boston MA 02110-1301, USA.
+- *****************************************************************************/
+-
+-#ifdef HAVE_CONFIG_H
+-#   include "config.h"
+-#endif
+-
+-#include <stdint.h>
+-#include <assert.h>
+-
+-#include <vlc_common.h>
+-#include <vlc_cpu.h>
+-#include <vlc_picture.h>
+-#include <vlc_filter.h>
+-
+-#include "deinterlace.h" /* filter_sys_t  */
+-#include "common.h"      /* FFMIN3 et al. */
+-
+-#include "algo_yadif.h"
+-
+-/*****************************************************************************
+- * Yadif (Yet Another DeInterlacing Filter).
+- *****************************************************************************/
+-
+-/* yadif.h comes from yadif.c of FFmpeg project.
+-   Necessary preprocessor macros are defined in common.h. */
+-#include "yadif.h"
+-
+-int RenderYadifSingle( filter_t *p_filter, picture_t *p_dst, picture_t *p_src )
+-{
+-    return RenderYadif( p_filter, p_dst, p_src, 0, 0 );
+-}
+-
+-int RenderYadif( filter_t *p_filter, picture_t *p_dst, picture_t *p_src,
+-                 int i_order, int i_field )
+-{
+-    VLC_UNUSED(p_src);
+-
+-    filter_sys_t *p_sys = p_filter->p_sys;
+-
+-    /* */
+-    assert( i_order >= 0 && i_order <= 2 ); /* 2 = soft field repeat */
+-    assert( i_field == 0 || i_field == 1 );
+-
+-    /* As the pitches must match, use ONLY pictures coming from picture_New()! */
+-    picture_t *p_prev = p_sys->context.pp_history[0];
+-    picture_t *p_cur  = p_sys->context.pp_history[1];
+-    picture_t *p_next = p_sys->context.pp_history[2];
+-
+-    /* Account for soft field repeat.
+-
+-       The "parity" parameter affects the algorithm like this (from yadif.h):
+-       uint8_t *prev2= parity ? prev : cur ;
+-       uint8_t *next2= parity ? cur  : next;
+-
+-       The original parity expression that was used here is:
+-       (i_field ^ (i_order == i_field)) & 1
+-
+-       Truth table:
+-       i_field = 0, i_order = 0  => 1
+-       i_field = 1, i_order = 1  => 0
+-       i_field = 1, i_order = 0  => 1
+-       i_field = 0, i_order = 1  => 0
+-
+-       => equivalent with e.g.  (1 - i_order)  or  (i_order + 1) % 2
+-
+-       Thus, in a normal two-field frame,
+-             parity 1 = first field  (i_order == 0)
+-             parity 0 = second field (i_order == 1)
+-
+-       Now, with three fields, where the third is a copy of the first,
+-             i_order = 0  =>  parity 1 (as usual)
+-             i_order = 1  =>  due to the repeat, prev = cur, but also next = cur.
+-                              Because in such a case there is no motion
+-                              (otherwise field repeat makes no sense),
+-                              we don't actually need to invoke Yadif's filter().
+-                              Thus, set "parity" to 2, and use this to bypass
+-                              the filter.
+-             i_order = 2  =>  parity 0 (as usual)
+-    */
+-    int yadif_parity;
+-    if( p_cur  &&  p_cur->i_nb_fields > 2 )
+-        yadif_parity = (i_order + 1) % 3; /* 1, *2*, 0; where 2 is a special
+-                                             value meaning "bypass filter". */
+-    else
+-        yadif_parity = (i_order + 1) % 2; /* 1, 0 */
+-
+-    /* Filter if we have all the pictures we need */
+-    if( p_prev && p_cur && p_next )
+-    {
+-        /* */
+-        void (*filter)(uint8_t *dst, uint8_t *prev, uint8_t *cur, uint8_t *next,
+-                       int w, int prefs, int mrefs, int parity, int mode);
+-
+-/* android clang build for x86 fails as not enough registers are available */
+-#if !defined(__ANDROID__)
+-# if defined(HAVE_YADIF_SSSE3)
+-        if( vlc_CPU_SSSE3() )
+-            filter = yadif_filter_line_ssse3;
+-        else
+-# endif
+-# if defined(HAVE_YADIF_SSE2)
+-        if( vlc_CPU_SSE2() )
+-            filter = yadif_filter_line_sse2;
+-        else
+-# endif
+-# if defined(HAVE_YADIF_MMX)
+-        if( vlc_CPU_MMX() )
+-            filter = yadif_filter_line_mmx;
+-        else
+-# endif
+-#endif
+-            filter = yadif_filter_line_c;
+-
+-        if( p_sys->chroma->pixel_size == 2 )
+-            filter = yadif_filter_line_c_16bit;
+-
+-        for( int n = 0; n < p_dst->i_planes; n++ )
+-        {
+-            const plane_t *prevp = &p_prev->p[n];
+-            const plane_t *curp  = &p_cur->p[n];
+-            const plane_t *nextp = &p_next->p[n];
+-            plane_t *dstp        = &p_dst->p[n];
+-
+-            for( int y = 1; y < dstp->i_visible_lines - 1; y++ )
+-            {
+-                if( (y % 2) == i_field  ||  yadif_parity == 2 )
+-                {
+-                    memcpy( &dstp->p_pixels[y * dstp->i_pitch],
+-                                &curp->p_pixels[y * curp->i_pitch], dstp->i_visible_pitch );
+-                }
+-                else
+-                {
+-                    int mode;
+-                    /* Spatial checks only when enough data */
+-                    mode = (y >= 2 && y < dstp->i_visible_lines - 2) ? 0 : 2;
+-
+-                    assert( prevp->i_pitch == curp->i_pitch && curp->i_pitch == nextp->i_pitch );
+-                    filter( &dstp->p_pixels[y * dstp->i_pitch],
+-                            &prevp->p_pixels[y * prevp->i_pitch],
+-                            &curp->p_pixels[y * curp->i_pitch],
+-                            &nextp->p_pixels[y * nextp->i_pitch],
+-                            dstp->i_visible_pitch,
+-                            y < dstp->i_visible_lines - 2  ? curp->i_pitch : -curp->i_pitch,
+-                            y  - 1  ?  -curp->i_pitch : curp->i_pitch,
+-                            yadif_parity,
+-                            mode );
+-                }
+-
+-                /* We duplicate the first and last lines */
+-                if( y == 1 )
+-                    memcpy(&dstp->p_pixels[(y-1) * dstp->i_pitch],
+-                               &dstp->p_pixels[ y    * dstp->i_pitch],
+-                               dstp->i_pitch);
+-                else if( y == dstp->i_visible_lines - 2 )
+-                    memcpy(&dstp->p_pixels[(y+1) * dstp->i_pitch],
+-                               &dstp->p_pixels[ y    * dstp->i_pitch],
+-                               dstp->i_pitch);
+-            }
+-        }
+-
+-        p_sys->context.i_frame_offset = 1; /* p_cur will be rendered at next frame, too */
+-
+-        return VLC_SUCCESS;
+-    }
+-    else if( !p_prev && !p_cur && p_next )
+-    {
+-        /* NOTE: For the first frame, we use the default frame offset
+-                 as set by Open() or SetFilterMethod(). It is always 0. */
+-
+-        /* FIXME not good as it does not use i_order/i_field */
+-        RenderX( p_filter, p_dst, p_next );
+-        return VLC_SUCCESS;
+-    }
+-    else
+-    {
+-        p_sys->context.i_frame_offset = 1; /* p_cur will be rendered at next frame */
+-
+-        return VLC_EGENERIC;
+-    }
+-}
+diff --git a/modules/video_filter/deinterlace/algo_yadif.h b/modules/video_filter/deinterlace/algo_yadif.h
+deleted file mode 100644
+index 7bc4518b0..000000000
+--- a/modules/video_filter/deinterlace/algo_yadif.h
++++ /dev/null
+@@ -1,94 +0,0 @@
+-/*****************************************************************************
+- * algo_yadif.h : Wrapper for FFmpeg's Yadif algorithm
+- *****************************************************************************
+- * Copyright (C) 2000-2011 the VideoLAN team
+- * $Id$
+- *
+- * Author: Laurent Aimar <fenrir@videolan.org>
+- *         Juha Jeronen  <juha.jeronen@jyu.fi> (soft field repeat hack)
+- *
+- * This program is free software; you can redistribute it and/or modify it
+- * under the terms of the GNU Lesser General Public License as published by
+- * the Free Software Foundation; either version 2.1 of the License, or
+- * (at your option) any later version.
+- *
+- * This program is distributed in the hope that it will be useful,
+- * but WITHOUT ANY WARRANTY; without even the implied warranty of
+- * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+- * GNU Lesser General Public License for more details.
+- *
+- * You should have received a copy of the GNU Lesser General Public License
+- * along with this program; if not, write to the Free Software Foundation,
+- * Inc., 51 Franklin Street, Fifth Floor, Boston MA 02110-1301, USA.
+- *****************************************************************************/
+-
+-#ifndef VLC_DEINTERLACE_ALGO_YADIF_H
+-#define VLC_DEINTERLACE_ALGO_YADIF_H 1
+-
+-/**
+- * \file
+- * Adapter to fit the Yadif (Yet Another DeInterlacing Filter) algorithm
+- * from FFmpeg into VLC. The algorithm itself is implemented in yadif.h.
+- */
+-
+-/* Forward declarations */
+-struct filter_t;
+-struct picture_t;
+-
+-/*****************************************************************************
+- * Functions
+- *****************************************************************************/
+-
+-/**
+- * Yadif (Yet Another DeInterlacing Filter) from FFmpeg.
+- * One field is copied as-is (i_field), the other is interpolated.
+- *
+- * Comes with both interpolating and framerate doubling modes.
+- *
+- * If you do NOT want to use framerate doubling: use i_order = 0,
+- * and either 0 or 1 for i_field (keep it constant),
+- *
+- * If you DO want framerate doubling, do as instructed below.
+- *
+- * See Deinterlace() for usage examples of both modes.
+- *
+- * Needs three frames in the history buffer to operate.
+- * The first-ever frame is rendered using RenderX().
+- * The second is dropped. At the third frame, Yadif starts.
+- *
+- * Once Yadif starts, the frame that is rendered corresponds to the *previous*
+- * input frame (i_frame_offset = 1), complete with its original PTS.
+- * The latest input frame is used as the future/next frame, as reference
+- * for temporal interpolation.
+- *
+- * This wrapper adds support for soft field repeat (repeat_pict).
+- * Note that the generated "repeated" output picture is unique because
+- * of temporal interpolation.
+- *
+- * As many output frames should be requested for each input frame as is
+- * indicated by p_src->i_nb_fields. This is done by calling this function
+- * several times, first with i_order = 0, and then with all other parameters
+- * the same, but a new p_dst, increasing i_order (1 for second field,
+- * and then if i_nb_fields = 3, also i_order = 2 to get the repeated first
+- * field), and alternating i_field (starting, at i_order = 0, with the field
+- * according to p_src->b_top_field_first). See Deinterlace() for an example.
+- *
+- * @param p_filter The filter instance. Must be non-NULL.
+- * @param p_dst Output frame. Must be allocated by caller.
+- * @param p_src Input frame. Must exist.
+- * @param i_order Temporal field number: 0 = first, 1 = second, 2 = rep. first.
+- * @param i_field Keep which field? 0 = top field, 1 = bottom field.
+- * @return VLC error code (int).
+- * @retval VLC_SUCCESS The requested field was rendered into p_dst.
+- * @retval VLC_EGENERIC Frame dropped; only occurs at the second frame after start.
+- * @see Deinterlace()
+- */
+-int RenderYadif( filter_t *p_filter, picture_t *p_dst, picture_t *p_src,
+-                 int i_order, int i_field );
+-
+-/**
+- * Same as RenderYadif() but with no temporal references
+- */
+-int RenderYadifSingle( filter_t *p_filter, picture_t *p_dst, picture_t *p_src );
+-
+-#endif
+diff --git a/modules/video_filter/deinterlace/deinterlace.c b/modules/video_filter/deinterlace/deinterlace.c
+index 9f9d5291f..073424dcd 100644
+--- a/modules/video_filter/deinterlace/deinterlace.c
++++ b/modules/video_filter/deinterlace/deinterlace.c
+@@ -362,10 +362,6 @@ static struct filter_mode_t filter_mode [] = {
+                  { false, false, false, true }, true, true },
+     { "blend", .pf_render_single_pic = RenderBlend,
+                  { false, false, false, false }, true, true },
+-    { "yadif", .pf_render_single_pic = RenderYadifSingle,
+-                 { false, true, false, false }, false, true },
+-    { "yadif2x", .pf_render_ordered = RenderYadif,
+-                 { true, true, false, false }, false, true },
+     { "x", .pf_render_single_pic = RenderX,
+                  { false, false, false, false }, false, false },
+     { "phosphor", .pf_render_ordered = RenderPhosphor,
+diff --git a/modules/video_filter/deinterlace/deinterlace.h b/modules/video_filter/deinterlace/deinterlace.h
+index 355f2f488..e79629bb9 100644
+--- a/modules/video_filter/deinterlace/deinterlace.h
++++ b/modules/video_filter/deinterlace/deinterlace.h
+@@ -39,7 +39,6 @@ struct vlc_object_t;
+ /* Local algorithm headers */
+ #include "algo_basic.h"
+ #include "algo_x.h"
+-#include "algo_yadif.h"
+ #include "algo_phosphor.h"
+ #include "algo_ivtc.h"
+ #include "common.h"
+@@ -51,12 +50,12 @@ struct vlc_object_t;
+ /** Available deinterlace modes. */
+ static const char *const mode_list[] = {
+     "discard", "blend", "mean", "bob", "linear", "x",
+-    "yadif", "yadif2x", "phosphor", "ivtc" };
++    "phosphor", "ivtc" };
+ 
+ /** User labels for the available deinterlace modes. */
+ static const char *const mode_list_text[] = {
+     N_("Discard"), N_("Blend"), N_("Mean"), N_("Bob"), N_("Linear"), "X",
+-    "Yadif", "Yadif (2x)", N_("Phosphor"), N_("Film NTSC (IVTC)") };
++    N_("Phosphor"), N_("Film NTSC (IVTC)") };
+ 
+ /*****************************************************************************
+  * Data structures
+diff --git a/modules/video_filter/deinterlace/yadif.h b/modules/video_filter/deinterlace/yadif.h
+deleted file mode 100644
+index af16443d0..000000000
+--- a/modules/video_filter/deinterlace/yadif.h
++++ /dev/null
+@@ -1,153 +0,0 @@
+-/*
+- * Copyright (C) 2006 Michael Niedermayer <michaelni@gmx.at>
+- *
+- * This file is part of FFmpeg.
+- *
+- * FFmpeg is free software; you can redistribute it and/or modify
+- * it under the terms of the GNU General Public License as published by
+- * the Free Software Foundation; either version 2 of the License, or
+- * (at your option) any later version.
+- *
+- * FFmpeg is distributed in the hope that it will be useful,
+- * but WITHOUT ANY WARRANTY; without even the implied warranty of
+- * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+- * GNU General Public License for more details.
+- *
+- * You should have received a copy of the GNU General Public License along
+- * with FFmpeg; if not, write to the Free Software Foundation, Inc.,
+- * 51 Franklin Street, Fifth Floor, Boston, MA 02110-1301 USA.
+- */
+-
+-#ifdef HAVE_CONFIG_H
+-#   include "config.h"
+-#endif
+-
+-#include <stdalign.h>
+-
+-#if defined(__GNUC__)
+-# define ATTR_USED __attribute__((used))
+-#else
+-# define ATTR_USED
+-#endif
+-
+-typedef intptr_t x86_reg;
+-typedef struct { uint64_t a, b; } xmm_reg;
+-
+-const ATTR_USED alignas (16) xmm_reg pb_1 = {
+-    0x0101010101010101ULL, 0x0101010101010101ULL
+-};
+-const ATTR_USED alignas (16) xmm_reg pw_1 = {
+-    0x0001000100010001ULL, 0x0001000100010001ULL
+-};
+-
+-#ifdef CAN_COMPILE_SSSE3
+-#if defined(__SSE__) || defined(__GNUC__) || defined(__clang__)
+-// ================ SSSE3 =================
+-#define HAVE_YADIF_SSSE3
+-#define COMPILE_TEMPLATE_SSE 1
+-#define COMPILE_TEMPLATE_SSSE3 1
+-#define VLC_TARGET VLC_SSE
+-#define RENAME(a) a ## _ssse3
+-#include "yadif_template.h"
+-#undef COMPILE_TEMPLATE_SSE
+-#undef COMPILE_TEMPLATE_SSSE3
+-#undef VLC_TARGET
+-#undef RENAME
+-#endif
+-#endif
+-
+-#ifdef CAN_COMPILE_SSE2
+-#if defined(__SSE__) || defined(__GNUC__) || defined(__clang__)
+-// ================= SSE2 =================
+-#define HAVE_YADIF_SSE2
+-#define COMPILE_TEMPLATE_SSE 1
+-#define VLC_TARGET VLC_SSE
+-#define RENAME(a) a ## _sse2
+-#include "yadif_template.h"
+-#undef COMPILE_TEMPLATE_SSE
+-#undef VLC_TARGET
+-#undef RENAME
+-#endif
+-#endif
+-
+-#ifdef CAN_COMPILE_MMX
+-#if defined(__MMX__) || defined(__GNUC__) || defined(__clang__)
+-// ================ MMX =================
+-#define HAVE_YADIF_MMX
+-#define VLC_TARGET VLC_MMX
+-#define RENAME(a) a ## _mmx
+-#include "yadif_template.h"
+-#undef VLC_TARGET
+-#undef RENAME
+-#endif
+-#endif
+-
+-#define FFABS abs
+-
+-#define CHECK(j)\
+-    {   int score = FFABS(cur[mrefs-1+(j)] - cur[prefs-1-(j)])\
+-                  + FFABS(cur[mrefs  +(j)] - cur[prefs  -(j)])\
+-                  + FFABS(cur[mrefs+1+(j)] - cur[prefs+1-(j)]);\
+-        if (score < spatial_score) {\
+-            spatial_score= score;\
+-            spatial_pred= (cur[mrefs  +(j)] + cur[prefs  -(j)])>>1;\
+-
+-#define FILTER \
+-    for (x = 0;  x < w; x++) { \
+-        int c = cur[mrefs]; \
+-        int d = (prev2[0] + next2[0])>>1; \
+-        int e = cur[prefs]; \
+-        int temporal_diff0 = FFABS(prev2[0] - next2[0]); \
+-        int temporal_diff1 =(FFABS(prev[mrefs] - c) + FFABS(prev[prefs] - e) )>>1; \
+-        int temporal_diff2 =(FFABS(next[mrefs] - c) + FFABS(next[prefs] - e) )>>1; \
+-        int diff = FFMAX3(temporal_diff0>>1, temporal_diff1, temporal_diff2); \
+-        int spatial_pred = (c+e)>>1; \
+-        int spatial_score = FFABS(cur[mrefs-1] - cur[prefs-1]) + FFABS(c-e) \
+-                          + FFABS(cur[mrefs+1] - cur[prefs+1]) - 1; \
+- \
+-        CHECK(-1) CHECK(-2) }} }} \
+-        CHECK( 1) CHECK( 2) }} }} \
+- \
+-        if (mode < 2) { \
+-            int b = (prev2[2*mrefs] + next2[2*mrefs])>>1; \
+-            int f = (prev2[2*prefs] + next2[2*prefs])>>1; \
+-            int max = FFMAX3(d-e, d-c, FFMIN(b-c, f-e)); \
+-            int min = FFMIN3(d-e, d-c, FFMAX(b-c, f-e)); \
+- \
+-            diff = FFMAX3(diff, min, -max); \
+-        } \
+- \
+-        if (spatial_pred > d + diff) \
+-           spatial_pred = d + diff; \
+-        else if (spatial_pred < d - diff) \
+-           spatial_pred = d - diff; \
+- \
+-        dst[0] = spatial_pred; \
+- \
+-        dst++; \
+-        cur++; \
+-        prev++; \
+-        next++; \
+-        prev2++; \
+-        next2++; \
+-    }
+-
+-static void yadif_filter_line_c(uint8_t *dst, uint8_t *prev, uint8_t *cur, uint8_t *next, int w, int prefs, int mrefs, int parity, int mode) {
+-    int x;
+-    uint8_t *prev2= parity ? prev : cur ;
+-    uint8_t *next2= parity ? cur  : next;
+-    FILTER
+-}
+-
+-static void yadif_filter_line_c_16bit(uint8_t *dst8, uint8_t *prev8, uint8_t *cur8, uint8_t *next8, int w, int prefs, int mrefs, int parity, int mode) {
+-    uint16_t *dst = (uint16_t *)dst8;
+-    uint16_t *prev = (uint16_t *)prev8;
+-    uint16_t *cur = (uint16_t *)cur8;
+-    uint16_t *next = (uint16_t *)next8;
+-    int x;
+-    uint16_t *prev2= parity ? prev : cur ;
+-    uint16_t *next2= parity ? cur  : next;
+-    mrefs /= 2;
+-    prefs /= 2;
+-    FILTER
+-}
+diff --git a/modules/video_filter/deinterlace/yadif_template.h b/modules/video_filter/deinterlace/yadif_template.h
+deleted file mode 100644
+index df4897db3..000000000
+--- a/modules/video_filter/deinterlace/yadif_template.h
++++ /dev/null
+@@ -1,282 +0,0 @@
+-/*
+- * Copyright (C) 2006 Michael Niedermayer <michaelni@gmx.at>
+- *
+- * This file is part of FFmpeg.
+- *
+- * FFmpeg is free software; you can redistribute it and/or modify
+- * it under the terms of the GNU General Public License as published by
+- * the Free Software Foundation; either version 2 of the License, or
+- * (at your option) any later version.
+- *
+- * FFmpeg is distributed in the hope that it will be useful,
+- * but WITHOUT ANY WARRANTY; without even the implied warranty of
+- * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+- * GNU General Public License for more details.
+- *
+- * You should have received a copy of the GNU General Public License along
+- * with FFmpeg; if not, write to the Free Software Foundation, Inc.,
+- * 51 Franklin Street, Fifth Floor, Boston, MA 02110-1301 USA.
+- */
+-
+-/* For some reason clang doens't like that %%rip macro */
+-
+-#if defined(__x86_64__) && !defined(__APPLE__)
+-#define MANGLE(a) "" #a "(%%rip)"
+-#define MANGLEVARIABLES [mode] "g"(mode),
+-#else
+-#define MANGLE(a) "" "%["#a"]"
+-#define MANGLEVARIABLES [pw_1] "m"(pw_1),\
+-             [pb_1] "m"(pb_1),\
+-             [mode] "g"(mode),
+-#endif
+-
+-#ifdef COMPILE_TEMPLATE_SSE
+-#define REGMM "xmm"
+-#define MM "%%"REGMM
+-#define MOV  "movq"
+-#define MOVQ "movdqa"
+-#define MOVQU "movdqu"
+-#define STEP 8
+-#define LOAD(mem,dst) \
+-            MOV"       "mem", "dst" \n\t"\
+-            "punpcklbw "MM"7, "dst" \n\t"
+-#define PSRL1(reg) "psrldq $1, "reg" \n\t"
+-#define PSRL2(reg) "psrldq $2, "reg" \n\t"
+-#define PSHUF(src,dst) "movdqa "dst", "src" \n\t"\
+-                       "psrldq $2, "src"     \n\t"
+-#else
+-#define REGMM "mm"
+-#define MM "%%"REGMM
+-#define MOV  "movd"
+-#define MOVQ "movq"
+-#define MOVQU "movq"
+-#define STEP 4
+-#define LOAD(mem,dst) \
+-            MOV"       "mem", "dst" \n\t"\
+-            "punpcklbw "MM"7, "dst" \n\t"
+-#define PSRL1(reg) "psrlq $8, "reg" \n\t"
+-#define PSRL2(reg) "psrlq $16, "reg" \n\t"
+-#define PSHUF(src,dst) "pshufw $9, "dst", "src" \n\t"
+-#endif
+-
+-#ifdef COMPILE_TEMPLATE_SSSE3
+-#define PABS(tmp,dst) \
+-            "pabsw     "dst", "dst" \n\t"
+-#else
+-#define PABS(tmp,dst) \
+-            "pxor     "tmp", "tmp" \n\t"\
+-            "psubw    "dst", "tmp" \n\t"\
+-            "pmaxsw   "tmp", "dst" \n\t"
+-#endif
+-
+-
+-#define CHECK(pj,mj) \
+-            MOVQU" "#pj"(%[cur],%[mrefs]), "MM"2 \n\t" /* cur[x-refs-1+j] */\
+-            MOVQU" "#mj"(%[cur],%[prefs]), "MM"3 \n\t" /* cur[x+refs-1-j] */\
+-            MOVQ"      "MM"2, "MM"4 \n\t"\
+-            MOVQ"      "MM"2, "MM"5 \n\t"\
+-            "pxor      "MM"3, "MM"4 \n\t"\
+-            "pavgb     "MM"3, "MM"5 \n\t"\
+-            "pand      "MANGLE(pb_1)", "MM"4 \n\t"\
+-            "psubusb   "MM"4, "MM"5 \n\t"\
+-            PSRL1(MM"5")                 \
+-            "punpcklbw "MM"7, "MM"5 \n\t" /* (cur[x-refs+j] + cur[x+refs-j])>>1 */\
+-            MOVQ"      "MM"2, "MM"4 \n\t"\
+-            "psubusb   "MM"3, "MM"2 \n\t"\
+-            "psubusb   "MM"4, "MM"3 \n\t"\
+-            "pmaxub    "MM"3, "MM"2 \n\t"\
+-            MOVQ"      "MM"2, "MM"3 \n\t"\
+-            MOVQ"      "MM"2, "MM"4 \n\t" /* ABS(cur[x-refs-1+j] - cur[x+refs-1-j]) */\
+-            PSRL1(MM"3")                  /* ABS(cur[x-refs  +j] - cur[x+refs  -j]) */\
+-            PSRL2(MM"4")                  /* ABS(cur[x-refs+1+j] - cur[x+refs+1-j]) */\
+-            "punpcklbw "MM"7, "MM"2 \n\t"\
+-            "punpcklbw "MM"7, "MM"3 \n\t"\
+-            "punpcklbw "MM"7, "MM"4 \n\t"\
+-            "paddw     "MM"3, "MM"2 \n\t"\
+-            "paddw     "MM"4, "MM"2 \n\t" /* score */
+-
+-#define CHECK1 \
+-            MOVQ"      "MM"0, "MM"3 \n\t"\
+-            "pcmpgtw   "MM"2, "MM"3 \n\t" /* if(score < spatial_score) */\
+-            "pminsw    "MM"2, "MM"0 \n\t" /* spatial_score= score; */\
+-            MOVQ"      "MM"3, "MM"6 \n\t"\
+-            "pand      "MM"3, "MM"5 \n\t"\
+-            "pandn     "MM"1, "MM"3 \n\t"\
+-            "por       "MM"5, "MM"3 \n\t"\
+-            MOVQ"      "MM"3, "MM"1 \n\t" /* spatial_pred= (cur[x-refs+j] + cur[x+refs-j])>>1; */
+-
+-#define CHECK2 /* pretend not to have checked dir=2 if dir=1 was bad.\
+-                  hurts both quality and speed, but matches the C version. */\
+-            "paddw    "MANGLE(pw_1)", "MM"6 \n\t"\
+-            "psllw     $14,   "MM"6 \n\t"\
+-            "paddsw    "MM"6, "MM"2 \n\t"\
+-            MOVQ"      "MM"0, "MM"3 \n\t"\
+-            "pcmpgtw   "MM"2, "MM"3 \n\t"\
+-            "pminsw    "MM"2, "MM"0 \n\t"\
+-            "pand      "MM"3, "MM"5 \n\t"\
+-            "pandn     "MM"1, "MM"3 \n\t"\
+-            "por       "MM"5, "MM"3 \n\t"\
+-            MOVQ"      "MM"3, "MM"1 \n\t"
+-
+-#if defined(__MINGW32__) && defined(_WIN32) && !defined(_WIN64)
+-__attribute__((__force_align_arg_pointer__))
+-#endif
+-VLC_TARGET static void RENAME(yadif_filter_line)(uint8_t *dst,
+-                              uint8_t *prev, uint8_t *cur, uint8_t *next,
+-                              int w, int prefs, int mrefs, int parity, int mode)
+-{
+-    uint8_t tmpU[5*16];
+-    uint8_t *tmp= (uint8_t*)(((uintptr_t)(tmpU+15)) & ~15);
+-    int x;
+-
+-#define FILTER\
+-    for(x=0; x<w; x+=STEP){\
+-        __asm__ volatile(\
+-            "pxor      "MM"7, "MM"7 \n\t"\
+-            LOAD("(%[cur],%[mrefs])", MM"0") /* c = cur[x-refs] */\
+-            LOAD("(%[cur],%[prefs])", MM"1") /* e = cur[x+refs] */\
+-            LOAD("(%["prev2"])", MM"2") /* prev2[x] */\
+-            LOAD("(%["next2"])", MM"3") /* next2[x] */\
+-            MOVQ"      "MM"3, "MM"4 \n\t"\
+-            "paddw     "MM"2, "MM"3 \n\t"\
+-            "psraw     $1,    "MM"3 \n\t" /* d = (prev2[x] + next2[x])>>1 */\
+-            MOVQ"      "MM"0,   (%[tmp]) \n\t" /* c */\
+-            MOVQ"      "MM"3, 16(%[tmp]) \n\t" /* d */\
+-            MOVQ"      "MM"1, 32(%[tmp]) \n\t" /* e */\
+-            "psubw     "MM"4, "MM"2 \n\t"\
+-            PABS(      MM"4", MM"2") /* temporal_diff0 */\
+-            LOAD("(%[prev],%[mrefs])", MM"3") /* prev[x-refs] */\
+-            LOAD("(%[prev],%[prefs])", MM"4") /* prev[x+refs] */\
+-            "psubw     "MM"0, "MM"3 \n\t"\
+-            "psubw     "MM"1, "MM"4 \n\t"\
+-            PABS(      MM"5", MM"3")\
+-            PABS(      MM"5", MM"4")\
+-            "paddw     "MM"4, "MM"3 \n\t" /* temporal_diff1 */\
+-            "psrlw     $1,    "MM"2 \n\t"\
+-            "psrlw     $1,    "MM"3 \n\t"\
+-            "pmaxsw    "MM"3, "MM"2 \n\t"\
+-            LOAD("(%[next],%[mrefs])", MM"3") /* next[x-refs] */\
+-            LOAD("(%[next],%[prefs])", MM"4") /* next[x+refs] */\
+-            "psubw     "MM"0, "MM"3 \n\t"\
+-            "psubw     "MM"1, "MM"4 \n\t"\
+-            PABS(      MM"5", MM"3")\
+-            PABS(      MM"5", MM"4")\
+-            "paddw     "MM"4, "MM"3 \n\t" /* temporal_diff2 */\
+-            "psrlw     $1,    "MM"3 \n\t"\
+-            "pmaxsw    "MM"3, "MM"2 \n\t"\
+-            MOVQ"      "MM"2, 48(%[tmp]) \n\t" /* diff */\
+-\
+-            "paddw     "MM"0, "MM"1 \n\t"\
+-            "paddw     "MM"0, "MM"0 \n\t"\
+-            "psubw     "MM"1, "MM"0 \n\t"\
+-            "psrlw     $1,    "MM"1 \n\t" /* spatial_pred */\
+-            PABS(      MM"2", MM"0")      /* ABS(c-e) */\
+-\
+-            MOVQU" -1(%[cur],%[mrefs]), "MM"2 \n\t" /* cur[x-refs-1] */\
+-            MOVQU" -1(%[cur],%[prefs]), "MM"3 \n\t" /* cur[x+refs-1] */\
+-            MOVQ"      "MM"2, "MM"4 \n\t"\
+-            "psubusb   "MM"3, "MM"2 \n\t"\
+-            "psubusb   "MM"4, "MM"3 \n\t"\
+-            "pmaxub    "MM"3, "MM"2 \n\t"\
+-            PSHUF(MM"3", MM"2") \
+-            "punpcklbw "MM"7, "MM"2 \n\t" /* ABS(cur[x-refs-1] - cur[x+refs-1]) */\
+-            "punpcklbw "MM"7, "MM"3 \n\t" /* ABS(cur[x-refs+1] - cur[x+refs+1]) */\
+-            "paddw     "MM"2, "MM"0 \n\t"\
+-            "paddw     "MM"3, "MM"0 \n\t"\
+-            "psubw    "MANGLE(pw_1)", "MM"0 \n\t" /* spatial_score */\
+-\
+-            CHECK(-2,0)\
+-            CHECK1\
+-            CHECK(-3,1)\
+-            CHECK2\
+-            CHECK(0,-2)\
+-            CHECK1\
+-            CHECK(1,-3)\
+-            CHECK2\
+-\
+-            /* if(p->mode<2) ... */\
+-            MOVQ"    48(%[tmp]), "MM"6 \n\t" /* diff */\
+-            "cmpl      $2, %[mode] \n\t"\
+-            "jge       1f \n\t"\
+-            LOAD("(%["prev2"],%[mrefs],2)", MM"2") /* prev2[x-2*refs] */\
+-            LOAD("(%["next2"],%[mrefs],2)", MM"4") /* next2[x-2*refs] */\
+-            LOAD("(%["prev2"],%[prefs],2)", MM"3") /* prev2[x+2*refs] */\
+-            LOAD("(%["next2"],%[prefs],2)", MM"5") /* next2[x+2*refs] */\
+-            "paddw     "MM"4, "MM"2 \n\t"\
+-            "paddw     "MM"5, "MM"3 \n\t"\
+-            "psrlw     $1,    "MM"2 \n\t" /* b */\
+-            "psrlw     $1,    "MM"3 \n\t" /* f */\
+-            MOVQ"   (%[tmp]), "MM"4 \n\t" /* c */\
+-            MOVQ" 16(%[tmp]), "MM"5 \n\t" /* d */\
+-            MOVQ" 32(%[tmp]), "MM"7 \n\t" /* e */\
+-            "psubw     "MM"4, "MM"2 \n\t" /* b-c */\
+-            "psubw     "MM"7, "MM"3 \n\t" /* f-e */\
+-            MOVQ"      "MM"5, "MM"0 \n\t"\
+-            "psubw     "MM"4, "MM"5 \n\t" /* d-c */\
+-            "psubw     "MM"7, "MM"0 \n\t" /* d-e */\
+-            MOVQ"      "MM"2, "MM"4 \n\t"\
+-            "pminsw    "MM"3, "MM"2 \n\t"\
+-            "pmaxsw    "MM"4, "MM"3 \n\t"\
+-            "pmaxsw    "MM"5, "MM"2 \n\t"\
+-            "pminsw    "MM"5, "MM"3 \n\t"\
+-            "pmaxsw    "MM"0, "MM"2 \n\t" /* max */\
+-            "pminsw    "MM"0, "MM"3 \n\t" /* min */\
+-            "pxor      "MM"4, "MM"4 \n\t"\
+-            "pmaxsw    "MM"3, "MM"6 \n\t"\
+-            "psubw     "MM"2, "MM"4 \n\t" /* -max */\
+-            "pmaxsw    "MM"4, "MM"6 \n\t" /* diff= MAX3(diff, min, -max); */\
+-            "1: \n\t"\
+-\
+-            MOVQ" 16(%[tmp]), "MM"2 \n\t" /* d */\
+-            MOVQ"      "MM"2, "MM"3 \n\t"\
+-            "psubw     "MM"6, "MM"2 \n\t" /* d-diff */\
+-            "paddw     "MM"6, "MM"3 \n\t" /* d+diff */\
+-            "pmaxsw    "MM"2, "MM"1 \n\t"\
+-            "pminsw    "MM"3, "MM"1 \n\t" /* d = clip(spatial_pred, d-diff, d+diff); */\
+-            "packuswb  "MM"1, "MM"1 \n\t"\
+-\
+-            ::[prev] "r"(prev),\
+-             [cur]  "r"(cur),\
+-             [next] "r"(next),\
+-             [prefs]"r"((x86_reg)prefs),\
+-             [mrefs]"r"((x86_reg)mrefs),\
+-             MANGLEVARIABLES\
+-             [tmp]  "r"(tmp)\
+-        );\
+-        __asm__ volatile(MOV" "MM"1, %0" :"=m"(*dst));\
+-        dst += STEP;\
+-        prev+= STEP;\
+-        cur += STEP;\
+-        next+= STEP;\
+-    }
+-
+-    if (parity) {
+-#define prev2 "prev"
+-#define next2 "cur"
+-        FILTER
+-#undef prev2
+-#undef next2
+-    } else {
+-#define prev2 "cur"
+-#define next2 "next"
+-        FILTER
+-#undef prev2
+-#undef next2
+-    }
+-}
+-#undef STEP
+-#undef REGMM
+-#undef MM
+-#undef MOV
+-#undef MOVQ
+-#undef MOVQU
+-#undef PSHUF
+-#undef PSRL1
+-#undef PSRL2
+-#undef LOAD
+-#undef PABS
+-#undef CHECK
+-#undef CHECK1
+-#undef CHECK2
+-#undef FILTER
+-
+diff --git a/src/libvlc-module.c b/src/libvlc-module.c
+index 56342523f..c2d4026e3 100644
+--- a/src/libvlc-module.c
++++ b/src/libvlc-module.c
+@@ -354,12 +354,12 @@ static const char * const  ppsz_deinterlace_text[] = {
+     "Deinterlace method to use for video processing.")
+ static const char * const ppsz_deinterlace_mode[] = {
+     "auto", "discard", "blend", "mean", "bob",
+-    "linear", "x", "yadif", "yadif2x", "phosphor",
++    "linear", "x", "phosphor",
+     "ivtc"
+ };
+ static const char * const ppsz_deinterlace_mode_text[] = {
+     N_("Auto"), N_("Discard"), N_("Blend"), N_("Mean"), N_("Bob"),
+-    N_("Linear"), "X", "Yadif", "Yadif (2x)", N_("Phosphor"),
++    N_("Linear"), "X", N_("Phosphor"),
+     N_("Film NTSC (IVTC)")
+ };
+ 
+diff --git a/src/video_output/interlacing.c b/src/video_output/interlacing.c
+index f07ef8ef5..2155c2f96 100644
+--- a/src/video_output/interlacing.c
++++ b/src/video_output/interlacing.c
+@@ -47,8 +47,6 @@ static const char deinterlace_modes[][9]= {
+     "bob",
+     "linear",
+     "x",
+-    "yadif",
+-    "yadif2x",
+     "phosphor",
+     "ivtc",
+ };
+-- 
+2.54.0 (Apple Git-157)
+
diff --git a/libvlc/patches/0048-contrib-ffmpeg-do-not-build-the-GPL-2-aarch64-AC-3-D.patch b/libvlc/patches/0048-contrib-ffmpeg-do-not-build-the-GPL-2-aarch64-AC-3-D.patch
new file mode 100644
index 0000000..0d1679c
--- /dev/null
+++ b/libvlc/patches/0048-contrib-ffmpeg-do-not-build-the-GPL-2-aarch64-AC-3-D.patch
@@ -0,0 +1,72 @@
+From d5fd432e4e639b6fa1eb94dafd80b889e063bc0d Mon Sep 17 00:00:00 2001
+From: Small Bricktory <studio@smallbricktory.com>
+Date: Wed, 28 Oct 2026 11:35:02 -0700
+Subject: [PATCH] contrib: ffmpeg: do not build the GPL-2+ aarch64 AC-3 DSP
+ code
+
+FFmpeg's libavcodec/aarch64/ac3dsp_init_aarch64.c and ac3dsp_neon.S
+carry GPL-2+ headers but are built without --enable-gpl. Patch them out
+of the aarch64 build; the AC-3 DSP functions (AC-3 encoder only) use
+the C versions.
+---
+ .../avcodec-aarch64-no-gpl-ac3dsp.patch       | 35 +++++++++++++++++++
+ contrib/src/ffmpeg/rules.mak                  |  1 +
+ 2 files changed, 36 insertions(+)
+ create mode 100644 contrib/src/ffmpeg/avcodec-aarch64-no-gpl-ac3dsp.patch
+
+diff --git a/contrib/src/ffmpeg/avcodec-aarch64-no-gpl-ac3dsp.patch b/contrib/src/ffmpeg/avcodec-aarch64-no-gpl-ac3dsp.patch
+new file mode 100644
+index 000000000..800f6a81f
+--- /dev/null
++++ b/contrib/src/ffmpeg/avcodec-aarch64-no-gpl-ac3dsp.patch
+@@ -0,0 +1,35 @@
++avcodec/aarch64: do not build the GPL-2+ AC-3 DSP NEON code
++
++libavcodec/aarch64/ac3dsp_init_aarch64.c and ac3dsp_neon.S carry GPL-2+
++headers and are built without --enable-gpl. Drop them; AC-3 DSP falls
++back to the C functions (used by the AC-3 encoder).
++
++--- a/libavcodec/ac3dsp.c	2026-10-28 11:34:41
+++++ b/libavcodec/ac3dsp.c	2026-10-28 11:34:41
++@@ -389,9 +389,7 @@
++     c->downmix               = NULL;
++     c->downmix_fixed         = NULL;
++ 
++-#if ARCH_AARCH64
++-    ff_ac3dsp_init_aarch64(c);
++-#elif ARCH_ARM
+++#if ARCH_ARM
++     ff_ac3dsp_init_arm(c);
++ #elif ARCH_X86 && HAVE_X86ASM
++     ff_ac3dsp_init_x86(c);
++--- a/libavcodec/aarch64/Makefile	2026-10-28 11:34:41
+++++ b/libavcodec/aarch64/Makefile	2026-10-28 11:34:41
++@@ -1,5 +1,4 @@
++ # subsystems
++-OBJS-$(CONFIG_AC3DSP)                   += aarch64/ac3dsp_init_aarch64.o
++ OBJS-$(CONFIG_FDCTDSP)                  += aarch64/fdctdsp_init_aarch64.o
++ OBJS-$(CONFIG_FMTCONVERT)               += aarch64/fmtconvert_init.o
++ OBJS-$(CONFIG_H264CHROMA)               += aarch64/h264chroma_init_aarch64.o
++@@ -42,7 +41,6 @@
++ # subsystems
++ NEON-OBJS-$(CONFIG_AAC_DECODER)         += aarch64/sbrdsp_neon.o
++ NEON-OBJS-$(CONFIG_AAC_ENCODER)         += aarch64/aacencdsp_neon.o
++-NEON-OBJS-$(CONFIG_AC3DSP)              += aarch64/ac3dsp_neon.o
++ NEON-OBJS-$(CONFIG_FDCTDSP)             += aarch64/fdctdsp_neon.o
++ NEON-OBJS-$(CONFIG_FMTCONVERT)          += aarch64/fmtconvert_neon.o
++ NEON-OBJS-$(CONFIG_H264CHROMA)          += aarch64/h264cmc_neon.o
+diff --git a/contrib/src/ffmpeg/rules.mak b/contrib/src/ffmpeg/rules.mak
+index 6e4d66daa..de540370e 100644
+--- a/contrib/src/ffmpeg/rules.mak
++++ b/contrib/src/ffmpeg/rules.mak
+@@ -266,6 +266,7 @@ ifdef USE_FFMPEG
+ 	$(APPLY) $(SRC)/ffmpeg/0001-bring-back-XP-support.patch
+ 	$(APPLY) $(SRC)/ffmpeg/0011-avcodec-videotoolboxenc-disable-calls-on-unsupported.patch
+ 	$(APPLY) $(SRC)/ffmpeg/avcodec-fix-compilation-visionos.patch
++	$(APPLY) $(SRC)/ffmpeg/avcodec-aarch64-no-gpl-ac3dsp.patch
+ endif
+ ifdef USE_LIBAV
+ 	$(APPLY) $(SRC)/ffmpeg/libav_gsm.patch
+-- 
+2.54.0 (Apple Git-157)
+
-- 
2.54.0 (Apple Git-157)

